shimmingtoolbox/unwrap/skimage_unwrap.py

```diff
@@ -57,10 +57,22 @@ def skimage_unwrap(nii_wrapped_phase, mag=None, mask=None, threshold=None, fname
         nii_mask = nib.Nifti1Image(unwrap_mask, nii_wrapped_phase.affine, header=nii_wrapped_phase.header)
         nib.save(nii_mask, fname_save_mask)
 
-    # Unwrap the phase
-    ma_wrapped_phase = ma.array(wrapped_phase, mask=~unwrap_mask)
+    # Mask the wrapped phase
+    ma_wrapped_phase = ma.array(wrapped_phase, mask=~unwrap_mask.astype(bool))
+
+    singleton = False
+    if ma_wrapped_phase.ndim == 3 and ma_wrapped_phase.shape[2] == 1:
+        # If the phase is 3D with a single slice, remove the last dimension
+        ma_wrapped_phase = ma_wrapped_phase[:, :, 0]
+        singleton = True
```

Comment:
Why not add the line from if singleton here instead?

Reply:
I did an if else

Reply:
Done here: 18c7d78d82f45626ff350eda5b758557ffb63187